gyp/pylib/gyp/easy_xml.py

```diff
@@ -115,11 +116,10 @@ def WriteXmlIfChanged(content, path, encoding='utf-8', pretty=False,
   xml_string = XmlToString(content, encoding, pretty)
   if win32 and os.linesep != '\r\n':
     xml_string = xml_string.replace('\n', '\r\n')
-    
-  try:
-    xml_string = xml_string.encode(encoding)
-  except Exception:
-    xml_string = unicode(xml_string, 'latin-1').encode(encoding)
+  
+  default_encoding = locale.getdefaultlocale()[1]
+  if default_encoding != encoding.upper():
```

Comment:
I think it makes sense to "normalize" both sides, since AFAIK there are no two [Language Tags](https://www.iana.org/assignments/language-tags/language-tags.xhtml) that differ only by case.